src/main/groovy/jenkins/automation/utils/CheckmarxUtils.groovy

```diff
@@ -0,0 +1,239 @@
+ * <br/>&#x2014; <code>groupId</code>: required, specifies group ID, which is
+ *               actually the team ID, and which can be gotten by
+ *               browser-inspecting the 'Team' select box for an existing
+ *               Checkmarx job in the Jenkins UI
+ * <br/>&#x2014; <code>preset</code>: optional, defaults to "Default 2014" preset,
+ *               the ID of the Checkmarx preset configuration to use; this can be
+ *               gotten by browser-inspecting the 'Preset' select box for an existing
+ *               Checkmarx job in the Jenkins UI
+ * <br/>&#x2014; <code>useOwnServerCredentials</code>: optional, defaults to
+ *               <code>false</code>, specifies whether to use the global or per-job
+ *               Checkmarx server URL/credentials
+ * <br/>&#x2014; <code>serverUrl</code>: optional, specifies the Checkmarx server
+ *               URL to be used
+ * <br/>&#x2014; <code>username</code>: optional, specifies the Checkmarx username
+ * <br/>&#x2014; <code>password</code>: optional, specifies the Checkmarx password
+ * <br/>&#x2014; <code>excludeFolders</code>: optional, see checkmarxConfigDefaults
+ *               for the default, specifies the directories to exclude from scanning
+ * <br/>&#x2014; <code>filterPattern</code>: optional, see checkmarxConfigDefaults
+ *               for the default, specifies globs to filter out
+ * <br/>&#x2014; <code>incremental</code>: optional, defaults to <code>true</code>,
+ *               specifies whether scan should be incremental
+ * <br/>&#x2014; <code>fullScanCycle</code>: optional, defaults to <code>10</code>,
+ *               specifies how frequently to run a full scan when normally doing
+ *               incremental scans
+ * <br/>&#x2014; <code>comment</code>: optional, additional comments to be adding
+ *               to the scan results
+ * <br/>&#x2014; <code>vulnerabilityThresholdEnabled</code>: optional, defaults to
+ *               <code>true</code>, fail the build if the number of vulnerabilities
+ *               exceeds the thresholds set
+ * <br/>&#x2014; <code>highThresholdDefault</code>: optional, defaults to
+ *               <code>1</code>, sets the threshold for max number of 'high'
+ *               vulnerabilities
+ * <br/>&#x2014; <code>mediumThresholdDefault</code>: optional, defaults to
+ *               <code>2</code>, sets the threshold for max number of 'medium'
+ *               vulnerabilities
+ * <br/>&#x2014; <code>lowThresholdDefault</code>: optional, defaults to
+ *               <code>3</code>, set the threshold for max number of 'low'
+ *               vulnerabilities
+ * <br/>&#x2014; <code>osaEnabled</code>: optional, defaults to
+ *               <code>false</code>, define whether or not to run
+ *               OSA dependency scans
+ * <br/>&#x2014; <code>osaIncludePattern</code>: optional, defaults to
+ *               <code>\*\*\/osa_dependencies\/\*\*</code>, set the pattern that OSA
```

Comment:
Wondering if this is significant enough documentation for the OSA scan path.

Reply:
seems fine to me